modules/nextflow/src/main/groovy/nextflow/processor/TaskHandler.groovy

```diff
@@ -214,6 +218,31 @@ abstract class TaskHandler {
         return record
     }
 
+    void writeMetaFile() {
+        final record = [
+            hash: task.hash.toString(),
+            inputs: task.getInputFilesMap().collect { name, path ->
+                [
+                    name: name,
+                    path: path.toString(),
+                    predecessor: ConcreteDAG.getPredecessorHash(path)
+                ]
+            },
+            outputs: task.getOutputsByType(FileOutParam).values().flatten().collect { path ->
+                final file = path.toFile()
+
+                [
+                    name: path.name,
+                    path: path.toString(),
+                    size: file.size(),
+                    checksum: file.isFile() ? FilesEx.getChecksum(path) : null
+                ]
```

Comment:
@pditommaso I think this code doesn't work for object storage because they don't support `path.toFile()`. Is there a better way to get the file size and checksum?

Reply:
We talked, it'd be better to rely on the checksum provided by the cloud object storage when available and fallback to compute it otherwise. Checksum should be computed using an inputstream to avoid overloading the heap with big files

Reply:
Even better, it could be used an interface `EtagAware` and implement it in [AzFileAttributes](https://github.com/nextflow-io/nextflow/blob/38c8bd335d286ddc3907254b9d6817581f2a67d0/plugins/nf-azure/src/main/nextflow/cloud/azure/nio/AzFileAttributes.groovy#L37-L37) and [S3FielAttributes](https://github.com/nextflow-io/nextflow/blob/38c8bd335d286ddc3907254b9d6817581f2a67d0/plugins/nf-amazon/src/main/com/upplication/s3fs/S3FileAttributes.java#L48-L48) (google is a problem because, there's no a nextflow implementation for it)

Reply:
You want to implement a filesystem for Google cloud storage? 😄

Reply:
Not really, but I guess an hack can be found to do the same